src/commands/domain.rs

```diff
@@ -148,3 +135,196 @@ fn print_existing_domains(domains: &DomainsDomains) -> Result<()> {
+    let linked_project = configs.get_linked_project().await?;
+
+    ensure_project_and_environment_exist(&client, &configs, &linked_project).await?;
+
+    let project = get_project(&client, &configs, linked_project.project.clone()).await?;
+
+    let service = get_service(&linked_project, &project, service_name)?;
+
+    let spinner = (std::io::stdout().is_terminal() && !json)
+        .then(|| {
+            creating_domain_spiner(Some(format!(
+                "Creating custom domain for service {}{}...",
+                service.name,
+                port.map(|p| format!(" on port {}", p)).unwrap_or_default()
+            )))
+        })
+        .and_then(|s| s.ok());
+
+    let is_available = post_graphql::<queries::CustomDomainAvailable, _>(
+        &client,
+        configs.get_backboard(),
+        queries::custom_domain_available::Variables {
+            domain: domain.clone(),
+        },
+    )
+    .await?
+    .custom_domain_available
+    .available;
+
+    if !is_available {
+        bail!("Domain is not available:\n\t{}", domain);
+    }
+
+    let vars = mutations::custom_domain_create::Variables {
+        input: mutations::custom_domain_create::CustomDomainCreateInput {
+            domain: domain.clone(),
+            environment_id: linked_project.environment.clone(),
+            project_id: linked_project.project.clone(),
+            service_id: service.id.clone(),
+            target_port: port.map(|p| p as i64),
+        },
+    };
+
+    let response =
+        post_graphql::<mutations::CustomDomainCreate, _>(&client, configs.get_backboard(), vars)
+            .await?;
+
+    spinner.map(|s| s.finish_and_clear());
+
+    if json {
+        println!("{}", serde_json::to_string_pretty(&response)?);
+        return Ok(());
+    }
+
+    println!("Domain created: {}", response.custom_domain_create.domain);
+
+    if response.custom_domain_create.status.dns_records.is_empty() {
+        // This case should be impossible, but added error handling for safety.
+        //
+        // It can only occur if the backend is not returning the correct data,
+        // and in that case, the post_graphql call should have already errored.
+        bail!("No DNS records found. Please check the Railway dashboard for more information.");
+    }
+
+    println!(
+        "To finish setting up your custom domain, add the following DNS records to {}:\n",
+        &response.custom_domain_create.status.dns_records[0].zone
+    );
+
+    print_dns(response.custom_domain_create.status.dns_records);
+
+    println!("\nNote: if the Name is \"@\", the DNS record should be created for the root of the domain.");
+    println!("*DNS changes can take up to 72 hours to propagate worldwide.");
+
+    Ok(())
+}
+
+fn print_dns(
+    domains: Vec<
+        mutations::custom_domain_create::CustomDomainCreateCustomDomainCreateStatusDnsRecords,
+    >,
+) {
+    // I benchmarked this iter().fold() and it's faster than using 3x iter().map()
```

Comment:
I think this is fine, but just for future reference, readability is much more important here than performance. Especially considering the `domains` length will be size 1 99% of the time.

Reply:
Yeah, that makes sense. For `n=1`, the performance is entirely irrelevant.

Reply:
Even for `n=2`. The difference will be so small it is negligible. Unless thousands of domains are being processed, readability is what we should optimize for.

Reply:
I think it is fine as is, just mainly a note for the future